src/world.rs

```diff
@@ -722,6 +774,37 @@ impl World {
     /// Adds a resource with the default ID (`0`).
     ///
     /// If the resource already exists it will be overwritten.
+    ///
+    /// ## Difference between resources and components
+    ///
+    /// While components exist per entity, resources are like globals in the `World`.
+    /// Components are stored in component storages, which are resources themselves.
+    ///
+    /// Everything that is `Any + Send + Sync` can be a resource.
+    ///
+    /// ## Built-in resources
+    ///
+    /// There are two built-in resources:
+    ///
+    /// * `LazyUpdate` and
+    /// * `EntitiesRes`
+    ///
+    /// Both of them should only be fetched immutably, which is why
+    /// the latter one has a type def for convenience: `Entities` which
+    /// is just `Fetch<EntitiesRes>`. Both resources are special and need
+    /// to execute code at the end of the frame, which is done in `World::maintain`.
```

Comment:
The text is fine, but it reminds me the fact that calling `World::maintain` is not enforced. Is there any way to encode this in type system?

Reply:
I wouldn't know how, unless we change our architecture :(